src/List.mo

```diff
@@ -417,4 +418,23 @@ module {
   public func toVarArray<A>(xs : List<A>) : [var A] =
     Array.thaw<A>(toArray<A>(xs));
 
+  /// Create an iterator from a list.
+  public func toIter<A>(xs: List<A>) : Iter.Iter<A> {
+    var ix : Nat = 0;
+    let s = size<A>(xs);
+    var list = xs;
+    object {
+      public func next() : ?A {
+        if (ix >= s) {
+          return null
+        } else {
+          let popped = pop<A>(list);
+          list := popped.1;
+          ix += 1;
+          return popped.0
+        }
+      }
+    }
+  };
```

Comment:
Thanks, this is the way to go. Now this can be simplified substantially by just doing a pattern match on the captured state:
```suggestion
     var state = xs;
     object {
       public func next() : ?A =
         switch state {
           case (?(hd, tl)) { state := tl; ?hd };
           case _ null
         }
     }
  }
```

Reply:
Thank you for your kind instruction. It's certainly smarter that way. I have fixed the implementation.

Reply:
The only doubt that remains in me is about naming. There is only one other `toIter` function innn the library, all other iterators are named `vals`. @crusso what do you think?